__tests__/shared/didCommAndDataStoreWithCredentials.ts

```diff
@@ -0,0 +1,193 @@
+        keys: [
+          {
+            type: 'Ed25519',
+            kid: 'didcomm-receiverKey-1',
+            publicKeyHex: 'b162e405b6485eff8a57932429b192ec4de13c06813e9028a7cdadf0e2703636',
+            privateKeyHex:
+              '19ed9b6949cfd0f9a57e30f0927839a985fa699491886ebcdda6a954d869732ab162e405b6485eff8a57932429b192ec4de13c06813e9028a7cdadf0e2703636',
+            kms: 'local',
+          },
+        ],
+        services: [
+          {
+            id: 'msg2',
+            type: 'DIDCommMessaging',
+            serviceEndpoint: 'http://localhost:3002/messaging',
+          },
+        ],
+        provider: 'did:fake',
+        alias: 'receiver',
+      })
+      return true
+    })
+    afterAll(testContext.tearDown)
+
+    it('should send a message', async () => {
+      expect.assertions(2)
+
+      const message = {
+        type: 'test',
+        to: receiver.did,
+        from: sender.did,
+        id: 'test',
+        body: { hello: 'world' },
+      }
+      const packedMessage = await agent.packDIDCommMessage({
+        packing: 'authcrypt',
+        message,
+      })
+      await agent.sendDIDCommMessage({
+        messageId: '123',
+        packedMessage,
+        recipientDidUrl: receiver.did,
+      })
+
+      const messages = await agent.dataStoreORMGetMessagesCount({})
+      expect(messages).toEqual(1)
+      const vcs = await agent.dataStoreORMGetVerifiableCredentialsCount({})
+      expect(vcs).toEqual(0)
+    })
+
+    const vc = (creator: IIdentifier, proofFormat: string): Promise<VerifiableCredential> => {
+      return agent.createVerifiableCredential({
+        credential: {
+          issuer: { id: creator.did },
+          '@context': ['https://www.w3.org/2018/credentials/v1', 'https://veramo.io/contexts/profile/v1'],
+          type: ['VerifiableCredential', 'Profile'],
+          issuanceDate: new Date().toISOString(),
+          credentialSubject: {
+            name: 'No, 33',
+          },
+        },
+        save: false,
+        proofFormat: proofFormat,
+      })
+    }
+
+    const packed = (vc: VerifiableCredential): Promise<IPackedDIDCommMessage> => {
+      return agent.packDIDCommMessage({ packing: 'none', message: {
+        id: v4(),
+        type: 'w3c.vc',
+        from: sender.did,
+        to: receiver.did,
+        body: vc
+      }})
+    }
+    
+    it('should save LDS credential found inside DIDCommMessage', async () => {
+      expect.assertions(2)
+      const creator = await agent.didManagerGetOrCreate({ alias: 'messageCreator1', provider: 'did:ethr'})
+
+      const verifiableCredential = await vc(creator, 'lds')
+      const packedMessage = await packed(verifiableCredential)
+
+      await agent.sendDIDCommMessage({        
+        messageId: 'test-jwt-success',
+        packedMessage,
+        recipientDidUrl: sender.did,
+      })
+      const messages = await agent.dataStoreORMGetMessagesCount({})
+      expect(messages).toEqual(2)
```

Comment:
testing message flow by count is prone to some weird test failures when multiple tests produce messages.

It would be better to use message IDs, or something else to check if they get processed correctly.